Sources/StreamChat/ChatClient.swift

```diff
@@ -284,6 +284,24 @@ public class ChatClient {
             completion: { completion?($0) }
         )
     }
+    
+    /// Connects the client with the given user.
+    ///
+    /// - Parameters:
+    ///   - userInfo: The user info passed to `connect` endpoint.
+    ///   - tokenProvider: The closure used to retreive a token. Token provider will be used to establish the initial connection and also to obtain the new token when the previous one expires.
+    ///
+    /// - Throws: An error while communicating with the Stream API.
+    /// - Returns: A type representing the connected user and its state.
+    @available(iOS 13.0, *)
+    public func connectUser(userInfo: UserInfo, tokenProvider: @escaping TokenProvider) async throws -> ConnectedUser {
```

Comment:
The return should probably be discardable here, since customers might not want to do anything with it when connecting